File: codeforces/B/Buttons.cpp
// Problem: B. Buttons
// Contest: Codeforces - Codeforces Round 164 (Div. 2)
// URL: https://codeforces.com/problemset/problem/268/B
// Memory Limit: 256 MB
// Time Limit: 1000 ms

#include <algorithm>
#include <array>
#include <cmath>
#include <cstring>
#include <iostream>
#include <set>
#include <vector>
using namespace std;

int
main()
{
  int n;
  cin >> n;

  int ans = n;

  for (int i = 2; i <= n; ++i) {
    int tmp = (i * (n - i)) + 1;
    // cout << "Log: " << (tmp) << '\n';
    ans += tmp;
  }

  cout << ans << '\n';
  return 0;
}

File: codeforces/A/Buttons.hs
-- Problem: A. Buttons
-- Contest: Codeforces - Codeforces Round 893 (Div. 2)
-- URL: https://codeforces.com/contest/1858/problem/A
-- Memory Limit: 256 MB
-- Time Limit: 1000 ms

import           Control.Monad (replicateM_)

main :: IO ()
main = do
  t <- readLn :: IO Int
  replicateM_ t $ do
    [a, b, c] <- map read . words <$> getLine

    let ans = if even c
                then if a > b
                       then 1
                       else 2
                else if a < b
                       then 2
                       else 1

    putStrLn $ if ans == 1
                 then "First"
                 else "Second"
